gap/base/methods.gi

```diff
@@ -0,0 +1,84 @@
+#############################################################################
+##
+##  This file is part of recog, a package for the GAP computer algebra system
+##  which provides a collection of methods for the constructive recognition
+##  of groups.
+##
+##  This files's authors include Sergio Siccha.
+##
+##  Copyright of recog belongs to its developers whose names are too numerous
+##  to list here. Please refer to the COPYRIGHT file for details.
+##
+##  SPDX-License-Identifier: GPL-3.0-or-later
+##
+##
+##  Implementation of recog methods
+##
+#############################################################################
+
+# HACK so that we can still treat RecogMethod objects as records
+RECOG.MethodComponentMap := rec(stamp := "Stamp",
+                                method := "func",
+                                comment := "Comment");
+InstallMethod(\., [IsRecogMethod, IsObject],
+function(x, nr)
+    local name, map;
+    name := NameRNam(nr);
+    if IsBound\.(RECOG.MethodComponentMap, nr) then
+        return x!.(RECOG.MethodComponentMap.(name));
+    fi;
+    return x!.(name);
+end);
+
+# HACK treat records as RecogMethod objects
+InstallOtherMethod(Stamp, [IsRecord],
+function(r)
+    return r.stamp;
+end);
+
+InstallMethod(CallFuncList, [IsRecogMethod, IsList],
+{f, args} -> CallFuncList(f!.func, args));
+
+InstallOtherMethod(Comment, [IsRecord],
+function(r)
+    return r.comment;
+end);
+
+# TODO: DisplayString, PrintString
+InstallMethod(ViewString, [IsRecogMethod],
+function(m)
+    local s;
+    s := "<RecogMethod \"";
+    Append(s, Stamp(m));
+    Append(s, "\": ");
+    Append(s, ViewString(UnpackRecogMethod(m)));
+    Append(s, ">");
+    return s;
+end);
+
+InstallGlobalFunction(RecogMethod,
+function(stamp, comment, func)
```

Comment:
So there is no support for `validatesOrAlwaysValidInput` yet, which affects conversion of methods which set it.

That said, there isn't a *real* need to set `validatesOrAlwaysValidInput`, is there? It's not as if we are using it for anything. It's just a marker saying "someone went over this method and believes they have established that it is correctly validating everything". The one reason to have it as an explicitly variable is that it helps us to catch methods which don't have it yet?

Reply:
At the moment, we can use it to check which methods don't have it yet, yes. I think in the long run it should be set to `true` for every method so we can remove it. Should we rather turn it into a source comment?